pkg/canary/reader/reader.go

```diff
@@ -92,6 +93,19 @@ func NewReader(writer io.Writer,
 	interval time.Duration,
 ) *Reader {
 	h := http.Header{}
+
+	// http.DefaultClient will be used in the case that the connection to Loki is http or TLS without client certs.
+	httpClient := http.DefaultClient
+	if tlsConfig != nil {
+		// For the mTLS case, use a http.Client configured with the client side certificates.
+		rt, err := config.NewTLSRoundTripper(tlsConfig, caFile, func(tls *tls.Config) (http.RoundTripper, error) {
+			return &http.Transport{TLSClientConfig: tls}, nil
+		})
+		if err != nil {
+			panic(fmt.Sprintf("Failed to create HTTPS transport with TLS config: %v", err))
```

Comment:
I don't think we should be using `panic`. Can we handle this in a less destructive way?

Reply:
Good point, there's no reason for a stack trace to come out here. I've kept this as a fatal error that will use `os.Exit` instead.